source/val/ValidationState.cpp

```diff
@@ -251,20 +253,25 @@ vector<uint32_t> ValidationState_t::UnresolvedForwardIds() const {
 }
 
 bool ValidationState_t::IsDefinedId(uint32_t id) const {
-  return all_definitions_.find(Id{id}) != end(all_definitions_);
+  return all_definitions_.find(id) != end(all_definitions_);
 }
 
-const Id* ValidationState_t::FindDef(uint32_t id) const {
-  if (all_definitions_.count(Id{id}) == 0) {
+const Instruction* ValidationState_t::FindDef(uint32_t id) const {
+  if (all_definitions_.count(id) == 0) {
     return nullptr;
   } else {
     /// We are in a const function, so we cannot use defs.operator[]().
     /// Luckily we know the key exists, so defs_.at() won't throw an
     /// exception.
-    return &all_definitions_.at(id);
+    return all_definitions_.at(id);
   }
 }
 
+Instruction* ValidationState_t::FindDef(uint32_t id) {
+  return const_cast<Instruction*>(
```

Comment:
The const casts are smelly. 

The other FindDef is already a const function, so why do you need the `const_cast<const ValidationState_t*>` ?